src/algorithms/masked_gibbs.jl

```diff
@@ -31,7 +32,7 @@ function masked_gibbs!(
 
     # Create inverted masks to compute train log likelihood.
     inv_masks = compute_complementary_masks(
-        masks, num_neurons(model), model.max_time)
+        masks, num_neurons(model), model.max_time+0.000000001)
```

Comment:
What is the purpose of this?

Can we define a keyword argument `max_time_tolerance = 1e-8` or something?

Reply:
I think there was some strange behaviour where it would just slightly get confused about a spike that was on the edge of the max time and the whole thing would explode. This fixed it (though is admittedly janky)